kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/explorer/tree/TreeExplorerView.java

```diff
@@ -43,47 +43,39 @@
     public void init(final TreeExplorer presenter) {
         this.presenter = presenter;
         initWidget(uiBinder.createAndBindUi(this));
+
         tree.addSelectionHandler(selectionEvent -> {
-            final TreeItem item = selectionEvent.getSelectedItem();
-            final String uuid = (String) item.getUserObject();
-            presenter.onSelect(uuid);
+            try {
```

Comment:
Sounds a bit strange using a `try/catch` block here... it's like you're trying to workaround another bug hehe. isnt it?? :) Notice that when removing a shape in the canvas, the  tree explorer is throwing this exception and error appears on the logs, so something is not good here.

I think the error is due to you're missing to obtain the result for this call `tree.addSelectionHandler(....)`, because even removing the treeitem from the tree, the handler instance is still "alive" and your `selectionEvent` fires, but there is no tree instance...

If you realize, the call `tree.addSelectionHandler(...)` returns an instance of `HandlerRegistration`. You have to keep this instance for each tree item you're adding handlers, this way you can call the method `removeHandler()` for the given instance when removing a tree item. So once you add some handler on a tree item, keep the registration instance for it, and once removing the tree item, remove the registrations for each handler. 
I expect that you can remove this `try/catch` once removing the handler instances in memory....